Nodejs/Product/TestAdapter/PackageJsonTestDiscoverer.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.NodejsTools.Npm;
+using Microsoft.NodejsTools.TestAdapter.TestFrameworks;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace Microsoft.NodejsTools.TestAdapter
+{
+    [FileExtension(".json")]
+    [DefaultExecutorUri(NodejsConstants.PackageJsonExecutorUriString)]
+    public sealed class PackageJsonTestDiscoverer : ITestDiscoverer
+    {
+        public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discoveryContext, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        {
+            foreach (var source in sources)
+            {
+                // we're only interested in package.json files here.
+                if (PackageJsonFactory.IsPackageJsonFile(source))
+                {
+                    this.DiscoverTestFiles(source, logger, discoverySink);
+                }
+            }
+        }
+
+        private void DiscoverTestFiles(string packageJsonPath, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        {
+            logger.SendMessage(TestMessageLevel.Informational, $"Parsing '{packageJsonPath}'.");
+
+            var packageJson = PackageJsonFactory.Create(packageJsonPath);
+            if (string.IsNullOrEmpty(packageJson.TestRoot))
+            {
+                logger.SendMessage(TestMessageLevel.Informational, "No vsTestOptions|testRoot specified.");
+                return;
+            }
+
+            var workingDir = Path.GetDirectoryName(packageJsonPath);
+            var testFolderPath = Path.Combine(workingDir, packageJson.TestRoot);
+            TestFramework testFx = null;
+
+            foreach (var dep in packageJson.AllDependencies)
+            {
+                testFx = FrameworkDiscover.Intance.Get(dep.Name);
+                if (testFx != null)
+                {
+                    break;
+                }
+            }
+            testFx = testFx ?? FrameworkDiscover.Intance.Get("ExportRunner");
```

Comment:
>FrameworkDiscover.Intance.Get("ExportRunner") [](start = 31, length = 45)

Is there some reason to believe this isn't null?

Reply:
Yes, this is installed by us and doesn't require additional `npm` packages.